inference-engine/src/inference_engine/network_serializer.cpp

```diff
@@ -4,165 +4,265 @@
+    for (auto i : node->inputs()) {
+      auto source_output = i.get_source_output();
+      auto source_node = source_output.get_node();
+      auto current_node = i.get_node();
+
+      Edge e{};
+      e.from_layer = layer_ids[source_node];
+      e.from_port = source_node->get_input_size() + source_output.get_index();
+      e.to_layer = layer_ids[current_node];
+      e.to_port = i.get_index();
+      edges.push_back(e);
     }
+  }
+  std::sort(begin(edges), end(edges), [](const Edge &a, const Edge &b) -> bool {
+    return a.from_layer < b.from_layer;
+  });
+  return edges;
 }
 
-}  // namespace
-
-void SerializeV10(const std::string& xmlPath, const std::string& binPath,
-                  const InferenceEngine::ICNNNetwork& network) {
-    if (auto function = network.getFunction()) {
-        // A flag for serializing executable graph information (not complete IR)
-        bool execGraphInfoSerialization = true;
-
-        // go over all operations and check whether performance stat is set
-        for (const auto & op : function->get_ops()) {
-            auto & rtInfo = op->get_rt_info();
-            if (rtInfo.find(ExecGraphInfoSerialization::PERF_COUNTER) == rtInfo.end()) {
-                execGraphInfoSerialization = false;
-                break;
-            }
-        }
+ConstantAtributes dump_constant_data(std::vector<uint8_t> &bin,
+                                     const ngraph::op::Constant &c) {
+  ConstantAtributes attr;
+  const uint8_t *p = reinterpret_cast<const uint8_t *>(c.get_data_ptr());
+  attr.size = ngraph::shape_size(c.get_shape()) * c.get_element_type().size();
+  attr.offset = bin.size();
+  bin.insert(end(bin), p, p + attr.size);
+  return attr;
+}
+
+void ngfunction_2_irv10(pugi::xml_document &doc, std::vector<uint8_t> &bin,
+                        const ngraph::Function &f) {
+  pugi::xml_node netXml = doc.append_child("net");
+  netXml.append_attribute("name").set_value(f.get_friendly_name().c_str());
+  netXml.append_attribute("version").set_value("10");
+  pugi::xml_node layers = netXml.append_child("layers");
+
+  auto layer_ids = create_layer_ids(f);
+  for (auto node : f.get_ordered_ops()) {
+    // <layers>
+    pugi::xml_node layer = layers.append_child("layer");
+    layer.append_attribute("id").set_value(layer_ids[node.get()]);
+    layer.append_attribute("name").set_value(node->get_friendly_name().c_str());
+    layer.append_attribute("type").set_value(node->get_type_name());
+    // TODO: calculate real opset
+    layer.append_attribute("version").set_value("opset1");
 
-        if (execGraphInfoSerialization) {
-            pugi::xml_document doc;
-            FillXmlDocWithExecutionNGraph(network, doc);
+    // <layers/data>
+    pugi::xml_node data = layer.append_child("data");
+
+    // <layers/data> general atributes
+    XmlVisitor visitor{data};
+    try {
+      if (!node->visit_attributes(visitor)) {
+        #if 0 // TODO: need to decide if we want to throw on vistiation fail
```

Comment:
I think that we can throw an exception in this case, because for serialization and deserialization all nodes should implement visit_attribute method and just add visit_attributes to `ExecutionNode` class.

Reply:
Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/8610ed93d0e2349e5a1508a018a5efd9b697e7fa)